drafthorse/pdf.py

```diff
@@ -291,3 +258,80 @@ def _facturx_update_metadata_add_attachment(
         )
     metadata_txt_dict = _prepare_pdf_metadata_txt(pdf_metadata)
     pdf_filestream.add_metadata(metadata_txt_dict)
+
+
+def _extract_xml_info(xml_data):
+    """
+    Extract metadata and profile from XML further added to the PDF
+    :param xml_data: XML data
+    :return: Metadata and profile
+    """
+    INVOICE_TYPE_CODE = "380"
+
+    xml_etree = etree.fromstring(xml_data)
+    namespaces = xml_etree.nsmap
+
+    # get metadata
+    date_xpath = xml_etree.xpath(
+        "//rsm:ExchangedDocument/ram:IssueDateTime/udt:DateTimeString",
+        namespaces=namespaces,
+    )
+    date = date_xpath[0].text
+    date_format = date_xpath[0].attrib and date_xpath[0].attrib.get("format") or "102"
+    format_map = {
+        "102": "%Y%m%d",
+        "203": "%Y%m%d%H%M",
+    }
+    date_dt = datetime.strptime(date, format_map.get(date_format, format_map["102"]))
+    number_xpath = xml_etree.xpath(
+        "//rsm:ExchangedDocument/ram:ID", namespaces=namespaces
+    )
+    number = number_xpath[0].text
+    seller_xpath = xml_etree.xpath(
+        "//ram:ApplicableHeaderTradeAgreement/ram:SellerTradeParty/ram:Name",
+        namespaces=namespaces,
+    )
+    seller = seller_xpath[0].text
+
+    doc_type_xpath = xml_etree.xpath(
+        "//rsm:ExchangedDocument/ram:TypeCode", namespaces=namespaces
+    )
+    doc_type = doc_type_xpath[0].text
+    if doc_type != INVOICE_TYPE_CODE:
+        raise Exception(
+            "Invalid doc type! XML value for TypeCode shall be 380 for an invoice."
+        )
+
+    doc_type_name = "Invoice"
+    date_str = datetime.strftime(date_dt, "%Y-%m-%d")
+    pdf_metadata = {
+        "author": seller,
+        "keywords": "{}, {}".format(doc_type_name, "Factur-X"),
+        "title": "{}: {} {}".format(seller, doc_type_name, number),
+        "subject": "{} {} dated {} issued by {}".format(
+            doc_type_name, number, date_str, seller
+        ),
```

Comment:
This would be the first thing in this library to hardcode English language, not sure it's worth it to go down that part? If we just used the invoice number similarly to the title, we'd also not need to extract the date, saving on complexity.